ui/dashboards/src/components/Dashboard/Dashboard.stories.tsx

```diff
@@ -33,6 +50,143 @@ export default meta;
 
 type Story = StoryObj<typeof Dashboard>;
 
+const DEFAULT_ALL_DASHBOARD: DashboardResource = {
+  kind: 'Dashboard',
+  metadata: {
+    name: 'TestAll',
+    created_at: '2023-03-17T20:13:03.570631Z',
+    updated_at: '2023-03-17T20:13:03.570631Z',
+    version: 0,
+    project: 'testing',
+  },
+  spec: {
+    display: {
+      name: 'TestAll',
+    },
+    panels: {
+      TimeSeries: {
+        kind: 'Panel',
+        spec: {
+          display: {
+            name: 'TimeSeries',
+          },
+          plugin: {
+            kind: 'TimeSeriesChart',
+            spec: {
+              legend: {
+                position: 'Right',
+              },
+              queries: [
+                {
+                  kind: 'TimeSeriesQuery',
+                  spec: {
+                    plugin: {
+                      kind: 'PrometheusTimeSeriesQuery',
+                      spec: {
+                        query: 'up{instance=~"$instance"}',
+                      },
+                    },
+                  },
+                },
+              ],
+            },
+          },
+        },
+      },
+    },
+    layouts: [
+      {
+        kind: 'Grid',
+        spec: {
+          display: {
+            title: 'Panel Group',
+            collapse: {
+              open: true,
+            },
+          },
+          items: [
+            {
+              x: 0,
+              y: 0,
+              width: 10,
+              height: 8,
+              content: {
+                $ref: '#/spec/panels/TimeSeries',
+              },
+            },
+          ],
+        },
+      },
+    ],
+    variables: [
+      {
+        kind: 'ListVariable',
+        spec: {
+          name: 'instance',
+          display: {
+            name: 'Instance',
+            hidden: false,
+          },
+          allow_all_value: true,
+          allow_multiple: true,
+          default_value: ['$__all'],
+          plugin: {
+            kind: 'PrometheusLabelValuesVariable',
+            spec: {
+              label_name: 'instance',
+            },
+          },
+        },
+      },
+    ],
+    duration: '5m',
+  },
+};
+
+function formatProviderParameters(dashboardState: DashboardResource) {
+  return {
+    withDashboard: {
+      props: {
+        initialState: {
+          dashboardResource: dashboardState,
+        },
+        dashboardResource: dashboardState,
+      },
+    },
+    withDatasourceStore: {
+      props: {
+        dashboardResource: dashboardState,
+      },
+    },
+    withTemplateVariables: {
+      props: {
+        initialVariableDefinitions: dashboardState.spec.variables,
+      },
+    },
+    WithTimeRange: {
+      props: {
+        dashboardDuration: dashboardState.spec.duration,
+      },
+    },
+  };
+}
+
+export const ListVariableWithDefaultAll: Story = {
```

Comment:
Reproduction of the bug for easy testing in storybook.

Reply:
Cool usage of Storybook 👍